app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -1,20 +1,30 @@
 class Users::OmniauthCallbacksController < Devise::OmniauthCallbacksController
+
   def facebook
+    # TODO fix this
+    intent = request.env["omniauth.auth"]["intent"]
     @user = User.from_omniauth(request.env["omniauth.auth"])
-
+    # @user = User.from_omniauth(auth, current_user)
+    #puts auth
     if @user.persisted?
+      #if intent == "sign_in"
+        #  puts intent
+        # Don't create a new identity
+      if intent == "sign_up"
+        # sign up flow, such as:
+        User.add_friends(@user.token, @user.id)
```

Comment:
@sdsunjay if you've got a user, why are you calling `User.add_friends(@user.token, @user.id)`? why not `@user.add_friends` that way, you don't have to pass anything into the function and it can retrieve it from the model it's already working with.

Reply:
if i use `@user.add_friends` how would you access token and id in the model?

Reply:
``` ruby
  def add_friends
     do_something_with(self.token, self.id)
  end
```

Reply:
or more simply.

``` ruby
  def add_friends
     do_something_with(token, id)
  end
```

self is implicit in this case, you only need to do self. when writing to the variable, not reading it.